psi4/src/psi4/libfock/jk.cc

```diff
@@ -147,6 +147,17 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
 
         return std::shared_ptr<JK>(jk);
 
+    } else if (jk_type == "COSX") {
+        DFJCOSK * jk = new DFJCOSK(primary, auxiliary, options);
```

Comment:
I think your suggested one-liner is not exactly equivalent to the current code. The current code returns and object of type `JK` (not `DFJCOSK`):
```
        DFJCOSK * jk = new DFJCOSK(primary, auxiliary, options);
        return std::shared_ptr<JK>(jk);
```
We could call `std::make_shared<DFJCOSK>` and then cast the object to `JK`, but that doesn't seem definitively cleaner to me, and I'd worry about doing the cast correctly.

Reply:
I assumed that C++ would do the cast for us upon object return, because the signature _says_ the it returns `std::shared_ptr<JK>`. If that isn't true, then disregard the comment.

Reply:
You're right, the `jk` object is casted just fine on return. Change made.